cli/src/cmd/forge/script.rs

```diff
@@ -742,6 +538,168 @@ impl ScriptArgs {
         })
     }
+
+    fn send_transactions(&self, deployment_sequence: &mut ScriptSequence) -> eyre::Result<()> {
+        // The user wants to actually send the transactions
+        let mut local_wallets = vec![];
+        if let Some(wallets) = self.wallets.private_keys()? {
+            wallets.into_iter().for_each(|wallet| local_wallets.push(wallet));
+        }
+
+        if let Some(wallets) = self.wallets.interactives()? {
+            wallets.into_iter().for_each(|wallet| local_wallets.push(wallet));
+        }
+
+        if let Some(wallets) = self.wallets.mnemonics()? {
+            wallets.into_iter().for_each(|wallet| local_wallets.push(wallet));
+        }
+
+        if let Some(wallets) = self.wallets.keystores()? {
+            wallets.into_iter().for_each(|wallet| local_wallets.push(wallet));
+        }
+
+        // TODO: Add trezor and ledger support (supported in multiwallet, just need to
+        // add derivation + SignerMiddleware creation logic)
+        // foundry/cli/src/opts/mod.rs:110
+        if local_wallets.is_empty() {
+            panic!("Error accessing local wallet when trying to send onchain transaction, did you set a private key, mnemonic or keystore?")
+        }
+
+        let fork_url = self
+            .evm_opts
+            .fork_url
+            .as_ref()
+            .expect("No fork_url provided for onchain sending")
+            .clone();
+        let provider = Provider::try_from(&fork_url).expect("Bad fork_url provider");
+
+        let rt = RuntimeOrHandle::new();
+        let chain = rt.block_on(provider.get_chainid())?.as_u64();
+        let is_legacy =
+            self.legacy || Chain::try_from(chain).map(|x| Chain::is_legacy(&x)).unwrap_or_default();
+        local_wallets =
+            local_wallets.into_iter().map(|wallet| wallet.with_chain_id(chain)).collect();
+
+        // in case of --force-resume, we forgive the first nonce disparity of each from
+        let mut nonce_offset: BTreeMap<Address, U256> = BTreeMap::new();
+
+        // Iterate through transactions, matching the `from` field with the associated
+        // wallet. Then send the transaction. Panics if we find a unknown `from`
+        deployment_sequence
+            .clone()
+            .transactions
+            .range((deployment_sequence.index as usize)..)
+            .map(|tx| {
+                let from = into_legacy_ref(tx).from.expect("No from for onchain transaction!");
+                if let Some(wallet) =
+                    local_wallets.iter().find(|wallet| (**wallet).address() == from)
+                {
+                    let signer = SignerMiddleware::new(provider.clone(), wallet.clone());
+                    Ok((tx.clone(), signer))
+                } else {
+                    Err(eyre::eyre!(format!(
+                        "No associated wallet for `from` address: {:?}. Unlocked wallets: {:?}. \nMake sure you have loaded all the private keys including of the `--sender`.",
```

Comment:
nit: consistency/clarity

```suggestion
                        "No associated wallet for address: {:?}. Unlocked wallets: {:?}. \nMake sure you have loaded all the private keys including the one for --sender.",
```